sources/topology_optimization/design.f90

```diff
@@ -42,31 +56,43 @@ module design_module
 
   !> @brief The topology type, which is used to describe the designs in
   !! topology optimization
-  type :: design_t
+  type, extends(simulation_component_t) :: design_t
```

Comment:
`design_t` have been defined both here and in `sources/bkp_design.f90` That will not be possible to compile.